packages/core/src/tools/edit.ts

```diff
@@ -51,7 +51,8 @@ export function applyReplacement(
   if (oldString === '' && !isNewFile) {
     return currentContent;
   }
-  return currentContent.replaceAll(oldString, newString);
+  // Use split/join to ensure replacement
+  return currentContent.split(oldString).join(newString);
```

Comment:
this doesn't seem safe. it will behave differently if the oldString is at the start or end of the currentContent.